cinema/views.py

```diff
@@ -118,12 +154,18 @@ class OrderPagination(PageNumberPagination):
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
 
     def get_queryset(self):
         return Order.objects.filter(user=self.request.user)
```

Comment:
The `get_queryset` method should use `self.queryset` instead of directly querying `Order.objects`. This ensures consistency and allows for easier modifications if the queryset is changed in the future. Consider changing `return Order.objects.filter(user=self.request.user)` to `return self.queryset.filter(user=self.request.user)`.